### DISCUSSION TITLE
Does sqldiff support partitioning statements?

### DISCUSSION THREAD

**Discussion by hanjie666** [sqldef/sqldef#635] [Feature requests]
## Platform

* OS: Linux
* RDBMS: MySQL
* Version:https://[sqldef](https://sqldef.github.io/).github.io/

## --export output
CREATE TABLE `test111` (
    `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT COMMENT 'id',
    `biz_no` varchar(64) NOT NULL DEFAULT '' COMMENT '业务号',
    `host_app` varchar(64) NOT NULL DEFAULT '' COMMENT '宿主机APP',
    `date_created` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT '创建时间',
    `created_by` varchar(32) NOT NULL DEFAULT 'sys' COMMENT '创建人',
    `date_updated` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP COMMENT '修改时间',
    `updated_by` varchar(32) NOT NULL DEFAULT 'sys' COMMENT '修改人',
    PRIMARY KEY (`id`, `date_created`)
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4 ROW_FORMAT=DYNAMIC COMMENT='接口访问日志表'
partition BY RANGE (TO_DAYS(date_created))
(PARTITION p20240624 VALUES LESS THAN (739426) ENGINE = InnoDB,
 PARTITION p20240625 VALUES LESS THAN (739427) ENGINE = InnoDB,
 PARTITION p20240626 VALUES LESS THAN (739428) ENGINE = InnoDB,
 PARTITION p20240627 VALUES LESS THAN (739429) ENGINE = InnoDB,
 PARTITION p20240628 VALUES LESS THAN (739430) ENGINE = InnoDB,
 PARTITION p20240629 VALUES LESS THAN (739431) ENGINE = InnoDB
)": syntax error at position 770

## Input SQL
-- 接口访问日志表
CREATE TABLE `test111` (
    `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT COMMENT 'id',
    `biz_no` varchar(64) NOT NULL DEFAULT '' COMMENT '业务号',
    `host_app` varchar(64) NOT NULL DEFAULT '' COMMENT '宿主机APP',
    `date_created` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT '创建时间',
    `created_by` varchar(32) NOT NULL DEFAULT 'sys' COMMENT '创建人',
    `date_updated` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP COMMENT '修改时间',
    `updated_by` varchar(32) NOT NULL DEFAULT 'sys' COMMENT '修改人',
    PRIMARY KEY (`id`, `date_created`)
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4 ROW_FORMAT=DYNAMIC COMMENT='接口访问日志表'
partition BY RANGE (TO_DAYS(date_created))
(PARTITION p20240624 VALUES LESS THAN (739426) ENGINE = InnoDB,
 PARTITION p20240625 VALUES LESS THAN (739427) ENGINE = InnoDB,
 PARTITION p20240626 VALUES LESS THAN (739428) ENGINE = InnoDB,
 PARTITION p20240627 VALUES LESS THAN (739429) ENGINE = InnoDB,
 PARTITION p20240628 VALUES LESS THAN (739430) ENGINE = InnoDB,
 PARTITION p20240629 VALUES LESS THAN (739431) ENGINE = InnoDB
);

## Current output
-- 接口访问日志表
CREATE TABLE `test111` (
    `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT COMMENT 'id',
    `biz_no` varchar(64) NOT NULL DEFAULT '' COMMENT '业务号',
    `host_app` varchar(64) NOT NULL DEFAULT '' COMMENT '宿主机APP',
    `date_created` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT '创建时间',
    `created_by` varchar(32) NOT NULL DEFAULT 'sys' COMMENT '创建人',
    `date_updated` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP COMMENT '修改时间',
    `updated_by` varchar(32) NOT NULL DEFAULT 'sys' COMMENT '修改人',
    PRIMARY KEY (`id`, `date_created`)
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4 ROW_FORMAT=DYNAMIC COMMENT='接口访问日志表'
partition BY RANGE (TO_DAYS(date_created))
(PARTITION p20240624 VALUES LESS THAN (739426) ENGINE = InnoDB,
 PARTITION p20240625 VALUES LESS THAN (739427) ENGINE = InnoDB,
 PARTITION p20240626 VALUES LESS THAN (739428) ENGINE = InnoDB,
 PARTITION p20240627 VALUES LESS THAN (739429) ENGINE = InnoDB,
 PARTITION p20240628 VALUES LESS THAN (739430) ENGINE = InnoDB,
 PARTITION p20240629 VALUES LESS THAN (739431) ENGINE = InnoDB
);

**Comment by gfx** [comment]
Resolved in https://github.com/sqldef/sqldef/pull/1036
